src/main/java/com/unascribed/fabrication/mixin/c_tweaks/feather_falling_no_trample/MixinFarmBlock.java

```diff
@@ -0,0 +1,33 @@
+package com.unascribed.fabrication.mixin.c_tweaks.feather_falling_no_trample;
+
+import com.unascribed.fabrication.support.EligibleIf;
+import net.minecraft.block.Block;
+import net.minecraft.block.FarmlandBlock;
+import net.minecraft.enchantment.EnchantmentHelper;
+import net.minecraft.enchantment.Enchantments;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(FarmlandBlock.class)
+@EligibleIf(configEnabled="*.feather_falling_no_trample")
+public class MixinFarmBlock extends Block {
+
+    public MixinFarmBlock(Settings settings) {
+        super(settings);
+    }
+
+    @Inject(method="onLandedUpon", at=@At("HEAD"), cancellable = true)
+    public void onLandedUpon(World world, BlockPos pos, Entity entity, float distance, CallbackInfo ci) {
+        if (entity instanceof LivingEntity
+                && (EnchantmentHelper.getEquipmentLevel(Enchantments.FEATHER_FALLING,(LivingEntity) entity)>=1)){
+            super.onLandedUpon(world, pos, entity, distance);
```

Comment:
Why call super?

Reply:
![image](https://user-images.githubusercontent.com/6185037/105614793-87c58d00-5d80-11eb-9270-b76e2b8c828e.png)

...Ah. Hm.

Reply:
I guess this is fine. Still feel like Entity::fall is the right place to patch for this...

Reply:
i thought you'd have to in order not to create a recursive function, but i guess CallbackInfo prevents that?

Reply:
The way this looks when injected into a method is something like this:

```java
CallbackInfo ci = new CallbackInfo(...);
MixinFarmBlock.onLandedUpon(world, pos, entity, distance, ci);
if (ci.isCancelled()) return;
/* rest of method follows... */
```

You don't *need* to call super, generally, but in this case it's needed in order to have the entity take fall damage properly.